HIP/hip-859.md

```diff
@@ -0,0 +1,140 @@
+---
+hip: 859
+title: Add support for gasConsumed field in REST APIs related to contract results
+author: Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>, Stoyan Panayotov <stoyan.panayotov@limechain.tech>
+working-group: Steven Sheehy <steven.sheehy@swirldslabs.com>, Nana Essilfie-Conduah <nana@swirldslabs.com>, Danno Ferrin <danno.ferrin@swirldslabs.com>, David Bakin <david.bakin@swirldslabs.com>
+type: Standards Track
+category: Mirror
+needs-council-approval: Yes
+status: Draft
+created: 2024-01-17
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/860
+updated: 2024-01-22
+---
+
+## Abstract
+
+Currently, consensus nodes export a `gasUsed` field for each EVM impacting transaction and the mirror node persists and exposes this as `gas_used` on the contract result REST APIs. This field value is calculated using the gas refund policy applied in the consensus node.
```

Comment:
nit: Might be nice to list the transactions. Not all readers may be aware of the 3 types at the time f writing.
```suggestion
Currently, consensus nodes export a `gasUsed` field for each EVM impacting transaction ([ContractCall](https://github.com/hashgraph/hedera-protobufs/blob/v0.46.0/services/contract_call.proto#L43), [ContractCreate](https://github.com/hashgraph/hedera-protobufs/blob/v0.46.0/services/contract_create.proto#L88) and [EthereumTransaction](https://github.com/hashgraph/hedera-protobufs/blob/v0.46.0/services/ethereum_transaction.proto#L31)) and the mirror node persists and exposes this as `gas_used` on the contract result REST APIs. This field value is calculated using the gas refund policy applied in the consensus node.
```

Reply:
Added